exonum-light-client/src/test/java/com/exonum/client/ExonumHttpClientBlocksIntegrationTest.java

```diff
@@ -0,0 +1,347 @@
+        + "        'end': 288\n"
+        + "    },\n"
+        + "    'blocks': [ " + BLOCK_1_JSON + "," + BLOCK_2_JSON + "," + BLOCK_3_JSON + "],\n"
+        + "    'times': ['" + BLOCK_1_TIME + "','" + BLOCK_2_TIME + "','" + BLOCK_3_TIME + "']\n"
+        + "}\n";
+    server.enqueue(new MockResponse().setBody(mockResponse));
+
+    // Call
+    int blocksCount = MAX_BLOCKS_PER_REQUEST;
+    boolean skipEmpty = true;
+    long height = Long.MAX_VALUE;
+    boolean showTimes = true;
+    BlocksResponse response = exonumClient.getBlocks(blocksCount, skipEmpty, height, showTimes);
+
+    // Assert response
+    assertThat(response.getBlocks(), contains(BLOCK_1, BLOCK_2, BLOCK_3));
+    assertThat(response.getBlocksRangeStart(), is(6L));
+    assertThat(response.getBlocksRangeEnd(), is(288L));
+
+    // Assert request params
+    RecordedRequest recordedRequest = server.takeRequest();
+    assertThat(recordedRequest.getMethod(), is("GET"));
+    assertThat(recordedRequest.getPath(), startsWith(BLOCKS));
+    assertBlockRequestParams(recordedRequest, blocksCount, skipEmpty, height, showTimes);
+  }
+
+  @Test
+  void getBlocksNoTime() throws InterruptedException {
+    String mockResponse = "{\n"
+        + "    'range': {\n"
+        + "        'start': 6,\n"
+        + "        'end': 288\n"
+        + "    },\n"
+        + "    'blocks': [ " + BLOCK_1_JSON + "," + BLOCK_2_JSON + "," + BLOCK_3_JSON + "],\n"
+        + "    'times': null\n"
+        + "}\n";
+    server.enqueue(new MockResponse().setBody(mockResponse));
+
+    // Call
+    int blocksCount = MAX_BLOCKS_PER_REQUEST;
+    boolean skipEmpty = true;
+    long height = Long.MAX_VALUE;
+    boolean showTimes = false;
+    BlocksResponse response = exonumClient.getBlocks(blocksCount, skipEmpty, height, showTimes);
+
+    // Assert response
+    assertThat(response.getBlocks(),
+        contains(BLOCK_1_WITHOUT_TIME, BLOCK_2_WITHOUT_TIME, BLOCK_3_WITHOUT_TIME));
+    assertThat(response.getBlocksRangeStart(), is(6L));
+    assertThat(response.getBlocksRangeEnd(), is(288L));
+
+    // Assert request params
+    RecordedRequest recordedRequest = server.takeRequest();
+    assertThat(recordedRequest.getMethod(), is("GET"));
+    assertThat(recordedRequest.getPath(), startsWith(BLOCKS));
+    assertBlockRequestParams(recordedRequest, blocksCount, skipEmpty, height, showTimes);
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {Integer.MIN_VALUE, -1, 0, MAX_BLOCKS_PER_REQUEST + 1, Integer.MAX_VALUE})
+  void getBlocksWrongBlocksCount(int blocksCount) {
+    assertThrows(IllegalArgumentException.class,
+        () -> exonumClient.getBlocks(blocksCount, false, 1L, false));
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {Integer.MIN_VALUE, -1, 0, MAX_BLOCKS_PER_REQUEST + 1, Integer.MAX_VALUE})
+  void getLastBlocksWrongBlocksCount(int blocksCount) {
+    assertThrows(IllegalArgumentException.class,
+        () -> exonumClient.getLastBlocks(blocksCount, false, false));
```

Comment:
Such number of boolean flags makes me wonder if we can do better.

```java
enum BlockRequestOption {
  SKIP_EMPTY_BLOCKS,
  INCLUDE_COMMIT_TIME,
}

// Possible API options:
getLastBlocks(long blocksCount, BlockRequestOption options...)
// or
getLastBlocks(long blocksCount, EnumSet<BlockRequestOption> options)

// But what are defaults?
```

or

```java
// Or EmptyBlock…
enum BlockFilteringOption {
  SKIP_EMPTY,
  INCLUDE_EMPTY,
}

enum BlockRequestOption {
  NO_COMMIT_TIME,
  INCLUDE_COMMIT_TIME,
}
getLastBlocks(long blocksCount, BlockFilteringOption filteringOption, BlockRequestOption requestOption)
```

@MakarovS , @bullet-tooth WDYT? I’d personally add this as an issue and discuss separately from this PR. I lean slightly towards option (2), because in (1) we will have to pick and document defaults.

Reply:
Second option > current > first option
Second option is much better IMO as it's more clear

Reply:
ECR-2994